kivy/uix/recycleview/views.py

```diff
@@ -139,10 +139,12 @@ class RecycleDataAdapter(EventDispatcher):
     with this instance.
     '''
 
-    # internals
-    views = {}  # current displayed items
-    # items whose attrs, except for pos/size is still accurate
-    dirty_views = defaultdict(dict)
+    def __init__(self):
+        super(RecycleDataAdapter, self).__init__()
+        # internals
+        self.views = {}  # current displayed items
```

Comment:
Rather than removing class declarations, keep them, but adding the per instance dicts like you did here should fix the original issue.